app/src/main/java/org/wikipedia/readinglist/ReadingListItemView.kt

```diff
@@ -182,8 +196,76 @@ class ReadingListItemView : ConstraintLayout {
                     list?.let { callback?.onRemoveAllOffline(it) }
                     return true
                 }
+                R.id.menu_reading_list_export -> {
+                    AlertDialog.Builder(context)
+                        .setTitle(R.string.reading_list_export_dialog_title)
+                        .setMessage(R.string.reading_list_export_dialog_message)
+                        .setPositiveButton(R.string.reading_list_remove_list_dialog_ok_button_text) { _, _ ->
+                            handlePermissionsAndExport()
+                        }
+                        .setNegativeButton(R.string.reading_list_remove_from_list_dialog_cancel_button_text, null)
+                        .create()
+                        .show()
+                    return true
+                }
                 else -> return false
             }
         }
     }
+
+    private fun handlePermissionsAndExport() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q || ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
+            exportListAsCsv()
+        } else {
+            activity.requestPermissionLauncher.launch(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        }
+    }
+
+    private fun exportListAsCsv() {
+        val stringBuilder = StringBuilder(context.getString(R.string.reading_list_csv_headers)).appendLine()
+        readingList?.let {
+            FeedbackUtil.showMessage(activity, R.string.reading_list_export_strated_message)
+            it.pages.forEach { page ->
+                val pageTitle = ReadingListPage.toPageTitle(page)
+                val uri = pageTitle.uri
+                val language = pageTitle.wikiSite.languageCode
+                stringBuilder.append(context.getString(R.string.reading_list_csv_entry, getSanitizedPageTitle(StringUtil.removeUnderscores(pageTitle.prefixedText)), language, uri)).appendLine()
+            }
+            val intent = Intent(DownloadManager.ACTION_VIEW_DOWNLOADS)
+            FileUtil.createFileInDownloadsFolder(context, "${readingList?.listTitle}.csv", stringBuilder)
+            context.getSystemService<NotificationManager>()?.notify(id, getNotificationBuilder(intent, readingList?.listTitle!!).build())
+            FeedbackUtil.showMessage(activity, R.string.reading_list_export_completed_message)
+        }
+    }
+
+    private fun getSanitizedPageTitle(title: String): String {
+        val sanitizedTitle: String = StringUtil.removeUnderscores(title)
+        return if (sanitizedTitle.contains(",")) {
+            context.getString(R.string.reading_list_csv_comma_title, sanitizedTitle)
+        } else {
+            sanitizedTitle
+        }
+    }
+
+    private fun getNotificationBuilder(intent: Intent, listName: String): NotificationCompat.Builder {
+        return NotificationCompat.Builder(context, NotificationCategory.SYSTEM.id)
```

Comment:
Hint: in order to make the notification "pop" on top of the screen (instead of the just the icon appearing silently), you need to use a notification category that is not `SYSTEM`. It needs to be another category that has an importance of `IMPORTANCE_HIGH`. We can co-opt an existing category like `MENTION`, or create a new category just for these notifications.

Reply:
Have repurposed `MENTION`, but wondering if we should keep it consistent with the system notification for media download, which is not a pop, but has a toast too in combination with a silent notification?